File: n383-ransom-note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        mag = [0] * 26
        
        for c in magazine:
            index = ord(c) - ord('a')
            mag[index] += 1
        
        for c in ransomNote:
            index = ord(c) - ord('a')
            mag[index] -= 1
            if mag[index] < 0:
                return False
        
        return True

File: n383-ransom-note.go
package main

func canConstruct(ransomNote string, magazine string) bool {
	a := [26]int{}

	for _, v := range []rune(magazine) {
		index := int(v - 'a')
		a[index]++
	}

	for _, v := range []rune(ransomNote) {
		index := int(v - 'a')
		a[index]--
		if a[index] < 0 {
			return false
		}
	}

	return true
}
